src/passes/MemoryPacking.cpp

```diff
@@ -125,20 +125,25 @@ struct MemoryPacking : public Pass {
     module->memory.segments.swap(packed);
   }
 
-  void optimizeTrappingBulkMemoryOps(PassRunner* runner, Module* module) {
-    struct Trapper : WalkerPass<PostWalker<Trapper>> {
+  void optimizeBulkMemoryOps(PassRunner* runner, Module* module) {
+    struct Nullifier : WalkerPass<PostWalker<Nullifier>> {
       bool isFunctionParallel() override { return true; }
       bool changed;
 
-      Pass* create() override { return new Trapper; }
+      Pass* create() override { return new Nullifier; }
 
       void visitMemoryInit(MemoryInit* curr) {
         if (!getModule()->memory.segments[curr->segment].isPassive) {
           Builder builder(*getModule());
           replaceCurrent(builder.blockify(builder.makeDrop(curr->dest),
                                           builder.makeDrop(curr->offset),
-                                          builder.makeDrop(curr->size),
-                                          builder.makeUnreachable()));
+                                          builder.makeDrop(curr->size)));
```

Comment:
This still needs to trap if offset or size are not equal to 0 or if the dest address is out of bounds. I'm actively working on my MemoryPacking patch again, but in the interim changing this code to the following should make the code correct again.

```
          // trap if (dest > memory.size | offset | size) != 0
          replaceCurrent(builder.makeIf(
            builder.makeBinary(
              OrInt32,
              builder.makeBinary(
                GtUInt32, curr->dest, builder.makeHost(MemorySize, Name(), {})),
              builder.makeBinary(OrInt32, curr->offset, curr->size)),
            builder.makeUnreachable()));
```

Reply:
Thanks!